### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Improve README

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

**Review by gravesm — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
-
-```tox -e refresh_modules --target-dir /somewhere/else```
+The `amazon_cloud_code_generator`'s core content is hosted in the `amazon_cloud_code_generator/` directory containing the following subdirectories:
+
+- `cmd/` - hosts the main logic of the `amazon_cloud_code_generator`.
+    ```
+    │   ├── generator.py  # Implements the logic to scrape the CloudFormation resource schema and performs all the pre-processing needed to produce each module's DOCUMENTATION block
+    │   ├── refresh_examples.py  # Implements the logic to update EXAMPLES block scraping all the tasks tagged with `tags: [docs]` from `amazon_cloud_code_generator/data/tests/integration/targets/`
+    │   ├── refresh_modules.py  # Implements the logic to generate all the modules specified inside `resources.py` including all the utilities
+    │   ├── resources.py  # Sets the AWS resources names we are going to generate
```
Comment (jillr) [inline README.md:22]:
```suggestion
    │   ├── resources.py  # Sets the AWS resource names we are going to generate  
```

File: README.md [hunk 2]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
-```tox -e refresh_modules --target-dir /somewhere/else```
+The `amazon_cloud_code_generator`'s core content is hosted in the `amazon_cloud_code_generator/` directory containing the following subdirectories:
+
+- `cmd/` - hosts the main logic of the `amazon_cloud_code_generator`.
+    ```
+    │   ├── generator.py  # Implements the logic to scrape the CloudFormation resource schema and performs all the pre-processing needed to produce each module's DOCUMENTATION block
+    │   ├── refresh_examples.py  # Implements the logic to update EXAMPLES block scraping all the tasks tagged with `tags: [docs]` from `amazon_cloud_code_generator/data/tests/integration/targets/`
+    │   ├── refresh_modules.py  # Implements the logic to generate all the modules specified inside `resources.py` including all the utilities
+    │   ├── resources.py  # Sets the AWS resources names we are going to generate  
+    │   ├── utils.py  # Implements utilities functions for the generator
```
Comment (jillr) [inline README.md:23]:
```suggestion
    │   ├── utils.py  # Implements utility functions for the generator
```

File: README.md [hunk 3]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    ```
+
+- `config/` - Contains the `modules.yaml` file where for each of the AWS resources we generate an entry representing the module's name must be added. Additionally, you can also set other information that will be used in module's DOCUMENTATION like short_description, description, and so forth.
```
Comment (jillr) [inline README.md:26]:
```suggestion
- `config/` - Contains the `modules.yaml` file where, for each of the AWS resources we generate, an entry representing the module's name must be added. Additionally, you can also set other information that will be used in module's DOCUMENTATION like short_description, description, and so forth.
```

File: README.md [hunk 4]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    │       │       ├── ...
+    │       └── unit
+    │           ├── constraints.txt
+    │           ├── module_utils
+    │           │   ├── test_core.py
+    │           │   └── test_utils.py
+    │           └── requirements.txt
+    ```
+
+- `templates/` - Hosts the templates files that will be used to generate each collection's plugin.
```
Comment (jillr) [inline README.md:55]:
```suggestion
- `templates/` - Hosts the templates files that will be used to generate each resource's plugin. 
```
I'm not sure about this one, but I think that you don't mean collection?

File: README.md [hunk 5]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    │           └── requirements.txt
+    ```
+
+- `templates/` - Hosts the templates files that will be used to generate each collection's plugin. 
+    ```
+    ├── default_module.j2
+    └── header.j2
+    ```
+
+The modules generated by the `amazon_cloud_code_generator/` will be made availabale in the `cloud` subdirectory by default. You can use the following command to do so:
```
Comment (jillr) [inline README.md:61]:
```suggestion
The modules generated by the `amazon_cloud_code_generator/` will be made available in the `cloud` subdirectory by default. You can use the following command to do so:
```

File: README.md [hunk 6]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
 
-Fork the original [``amazon.cloud collection``](https://github.com/ansible-collections/amazon.cloud) repository to your account on GitHub. Clone the repository from your fork:
+Fork the original [``amazon.cloud collection``](https://github.com/ansible-collections/amazon.cloud) repository to your account on GitHub. Ensure to have the path `~/.ansible/collections/ansible_collections/amazon/cloud` and clone the repository from your fork:
 ```
 mkdir -p ~/.ansible/collections/ansible_collections/amazon/cloud
 git clone https://github.com/YOUR_USER/amazon.cloud ~/.ansible/collections/ansible_collections/amazon/cloud
 ```
 
-Refresh the content of the modules moving in the repository path:
+Refresh the content of the modules moving in the repository path using `tox -e refresh_modules`.
```
Comment (jillr) [inline README.md:81]: It's unclear to me what `moving in the repository path` means here.  Would that be something like, "Refresh the content of the modules by running `tox -e refresh_modules` from the root of the repository"?

File: README.md [hunk 7]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
 tox -e refresh_modules
 ```
-Format the Python code of the modules using the black formatter:
+
+You can also specify the next release version by passing the `--next-version` parameter as follows:
+```
+tox -e refresh_modules --next-version 0.2.0
+```
+
+Individual opperations are also allowed.
```
Comment (jillr) [inline README.md:94]:
```suggestion
Individual operations are also allowed. 
```

File: README.md [hunk 8]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+```
+
+Individual opperations are also allowed. 
+
+For example, you can only format the Python code of the plugins using the black formatter by running:
 
 ```tox -e black```
 
-Refresh the modules documentation localed in `~/.ansible/collections/ansible_collections/amazon/cloud/docs`.
+You can also refresh the plugins documentation localed in `~/.ansible/collections/ansible_collections/amazon/cloud/docs` just by running.
```
Comment (jillr) [inline README.md:100]:
```suggestion
You can also refresh the plugins documentation located in `~/.ansible/collections/ansible_collections/amazon/cloud/docs` just by running.
```

**Review by jillr — COMMENTED** [review]
Some copy edit notes, hopefully the docs folks will agree with me!

File: README.md (hunk 8 above)
Comment (alinabuzachis) [reply README.md]: Rephrased like this: ``Refresh the content of the modules by moving in the repository path `~/.ansible/collections/ansible_collections/amazon/cloud` and running `tox -e refresh_modules`.
``

File: README.md (hunk 6 above)
Comment (alinabuzachis) [reply README.md]: Rephrased like this: ``Refresh the content of the modules by moving in the repository path `~/.ansible/collections/ansible_collections/amazon/cloud` and running `tox -e refresh_modules`.
``

File: README.md [hunk 9]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
 ## Usage
 
-The [``amazon.cloud collection``](https://github.com/ansible-collections/amazon.cloud) modules are built using:
-
-```tox -e refresh_modules```
-
-The modules will be generated in the `cloud` subdirectory by default. A different target directory can be specified by using:
-
-```tox -e refresh_modules --target-dir /somewhere/else```
+The `amazon_cloud_code_generator`'s core content is hosted in the `amazon_cloud_code_generator/` directory containing the following subdirectories:
```
Comment (oraNod) [inline README.md:15]:
```suggestion
The `amazon_cloud_code_generator` core content is hosted in the `amazon_cloud_code_generator/` directory, which contains the following subdirectories:
```

File: README.md [hunk 10]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+
+- `cmd/` - hosts the main logic of the `amazon_cloud_code_generator`.
+    ```
+    │   ├── generator.py  # Implements the logic to scrape the CloudFormation resource schema and performs all the pre-processing needed to produce each module's DOCUMENTATION block
```
Comment (oraNod) [inline README.md:19]:
```suggestion
    │   ├── generator.py  # Implements the logic to scrape the CloudFormation resource schema and performs all the pre-processing to produce the DOCUMENTATION block for each module
```

File: README.md [hunk 11]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    │   ├── refresh_examples.py  # Implements the logic to update EXAMPLES block scraping all the tasks tagged with `tags: [docs]` from `amazon_cloud_code_generator/data/tests/integration/targets/`
+    │   ├── refresh_modules.py  # Implements the logic to generate all the modules specified inside `resources.py` including all the utilities
+    │   ├── resources.py  # Sets the AWS resource names we are going to generate  
+    │   ├── utils.py  # Implements utility functions for the generator
+    ```
+
+- `config/` - Contains the `modules.yaml` file where, for each of the AWS resources we generate, an entry representing the module's name must be added. Additionally, you can also set other information that will be used in module's DOCUMENTATION like short_description, description, and so forth.
+
+- `data/` - Contains the content that will be copied over the generated collection. A high level overview of the subdirectory's content is shown below.
```
Comment (oraNod) [inline README.md:28]:
```suggestion
- `data/` - Contains the content that is copied to the generated collection. A high level overview of the subdirectory content is shown below:
```

File: README.md [hunk 12]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    │       │       ├── ...
+    │       └── unit
+    │           ├── constraints.txt
+    │           ├── module_utils
+    │           │   ├── test_core.py
+    │           │   └── test_utils.py
+    │           └── requirements.txt
+    ```
+
+- `templates/` - Hosts the templates files that will be used to generate each resource's plugin.
```
Comment (oraNod) [inline README.md:55]:
```suggestion
- `templates/` - Hosts the templates files that generate each resource's plugin. 
```

File: README.md [hunk 13]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    │           └── requirements.txt
+    ```
+
+- `templates/` - Hosts the templates files that will be used to generate each resource's plugin. 
+    ```
+    ├── default_module.j2
+    └── header.j2
+    ```
+
+The modules generated by the `amazon_cloud_code_generator/` will be made available in the `cloud` subdirectory by default. You can use the following command to do so:
```
Comment (oraNod) [inline README.md:61]:
```suggestion
The modules generated by the `amazon_cloud_code_generator/` are available in the `cloud` subdirectory by default. You can use the following command to do so:
```

Comment (oraNod) [reply README.md:61]: IMO that last sentence is not very clear. Does that provide an example of how to invoke modules generated in the `cloud` subdirectory?

If so, then maybe change "You can use the following command to do so:" to something like "You can invoke modules in that subdirectory as in the following example command:"

File: README.md [hunk 14]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
 
-Fork the original [``amazon.cloud collection``](https://github.com/ansible-collections/amazon.cloud) repository to your account on GitHub. Clone the repository from your fork:
+Fork the original [``amazon.cloud collection``](https://github.com/ansible-collections/amazon.cloud) repository to your account on GitHub. Ensure to have the path `~/.ansible/collections/ansible_collections/amazon/cloud` and clone the repository from your fork:
 ```
 mkdir -p ~/.ansible/collections/ansible_collections/amazon/cloud
 git clone https://github.com/YOUR_USER/amazon.cloud ~/.ansible/collections/ansible_collections/amazon/cloud
 ```
 
-Refresh the content of the modules moving in the repository path:
+Refresh the content of the modules by moving in the repository path `~/.ansible/collections/ansible_collections/amazon/cloud` and running `tox -e refresh_modules`.
```
Comment (oraNod) [inline README.md:81]:
```suggestion
Refresh the content of the modules by changing your path to `~/.ansible/collections/ansible_collections/amazon/cloud` and running `tox -e refresh_modules`.
```

File: README.md [hunk 15]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+```
+
+Individual operations are also allowed. 
+
+For example, you can only format the Python code of the plugins using the black formatter by running:
 
 ```tox -e black```
 
-Refresh the modules documentation localed in `~/.ansible/collections/ansible_collections/amazon/cloud/docs`.
+You can also refresh the plugins documentation located in `~/.ansible/collections/ansible_collections/amazon/cloud/docs` just by running.
```
Comment (oraNod) [inline README.md:100]:
```suggestion
You can also refresh the plugins documentation located in `~/.ansible/collections/ansible_collections/amazon/cloud/docs` by running.
```

File: README.md (hunk 11 above)
Comment (GomathiselviS) [inline README.md:28]:
```suggestion
- `data/` - Contains the content that will be copied over to the generated collection. A high level overview of the subdirectory's content is shown below.
```

File: README.md [hunk 16]
```diff
@@ -12,38 +12,92 @@ The `amazon_cloud_code_generator` relies on the CloudFormation client. Hence, th
...
+    │   ├── refresh_modules.py  # Implements the logic to generate all the modules specified inside `resources.py` including all the utilities
+    │   ├── resources.py  # Sets the AWS resource names we are going to generate  
+    │   ├── utils.py  # Implements utility functions for the generator
+    ```
+
+- `config/` - Contains the `modules.yaml` file where, for each of the AWS resources we generate, an entry representing the module's name must be added. Additionally, you can also set other information that will be used in module's DOCUMENTATION like short_description, description, and so forth.
+
+- `data/` - Contains the content that will be copied over the generated collection. A high level overview of the subdirectory's content is shown below.
+    ```
+    ├── changelogs  # Changelog fragments that will be used for the new collection release
```
Comment (GomathiselviS) [inline README.md:30]: I think changelogs and plugins folders should be at the same level.

Comment (GomathiselviS) [reply README.md:30]: I do not see the 'changelogs' folder under data. Am I missing something?

**Review by GomathiselviS — DISMISSED** [review]

File: README.md (hunk 13 above)
Comment (alinabuzachis) [reply README.md]: @oraNod Tried to rephrase. That command allows you to use the amazon_cloud_code_generator basically.

File: README.md (hunk 16 above)
Comment (alinabuzachis) [reply README.md]: I fixed the indentation level. You're right. changelogs folder was one my local copy. Just removed from the tree.

**Review by jillr — APPROVED** [review]

**Review by oraNod — APPROVED** [review]

**Pull request merged**